impeller/renderer/backend/metal/context_mtl.mm

```diff
@@ -392,6 +393,7 @@ new ContextMTL(device, command_queue,
 }
 
 void ContextMTL::FlushTasksAwaitingGPU() {
+  std::scoped_lock lock(tasks_awaiting_gpu_mutex_);
```

Comment:
It is now a deadlock to call StoreTasksForGPU from the callbacks. I think we should make this deadlock safe.

Reply:
You can create an empty container before obtaining the lock, swap it in the lock, and flush without the lock held. Ditto for the failure tasks up above.

Reply:
Done.